packages/cells/src/model.ts

```diff
@@ -284,6 +287,63 @@ export class CellModel extends CodeEditor.Model implements ICellModel {
     /* no-op */
   }
 
+  /**
+   * Handle a change to the cell metadata.
+   */
+  protected onModelDBMetadataChange(
+    sender: IObservableJSON,
+    event: IObservableJSON.IChangedArgs
+  ): void {
+    const metadata = this.nbcell.getMetadata();
+    this._mutex(() => {
+      switch (event.type) {
+        case 'add':
+          metadata[event.key] = event.newValue as any;
```

Comment:
Is this still compatible with `nbformat`? I think the `metadata` modeldb object describes some fields that do not exist in `nbformat` and vice-versa. E.g. `modeldb.collapsed` needs to be synced with `nbformat.jupyter.outputs_hidden`.

What do you think about the `trusted` flag? Should we make it part of the Yjs model or part of the unshared local model?

Reply:
Great question. The exact same question jumped to my min while I was writing that line. I tough, hey, let's see what it gives and if the attribute is not supported, the ui should complain... Then for the reverse sync (modelDB to shared), this is what I had to write. Apart from the way we mutate the object, the general question is "should we support non nbformat fields" I guess, yes, and at least the  `slide_type`attribute.

```
        if (newValue?.collapsed)
          this.metadata.set('collapsed', newValue?.collapsed);
        if (newValue?.jupyter) this.metadata.set('jupyter', newValue?.jupyter);
        if (newValue?.name) this.metadata.set('name', newValue?.name);
        if (newValue?.scrolled)
          this.metadata.set('scrolled', newValue?.scrolled);
        if (newValue?.tags) this.metadata.set('tags', newValue?.tags);
        if (newValue?.trusted) this.metadata.set('trusted', newValue?.trusted);
```

Reply:
> E.g. modeldb.collapsed needs to be synced with nbformat.jupyter.outputs_hidden.

So one good reason to be very explicit on  how we map fields between modeldb and nbmodel. If OK for you I can write something that would explicitly handle fields (so one line of code by field).

Reply:
My last commit takes the previous discussion into account.

Reply:
I added a few comments below about whether we should mix metadata with custom attributes. 

Where is `slide_type` usually stored. Why does it need to be part of nbmodel's metadata? We can also create custom fields for things with different restrictions. In general, I don't understand why there should be something like `slide_type` that is synced between clients, but that is not stored in the Notebook-JSON format. So in any case we are going to lose that information when the document is reloaded from the filesystem. So why do we need to keep it in collaborative sessions?